dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -1173,12 +1168,11 @@ private String getTopicClassificationTermOrTermAndVocabulary(DatasetField topicC
         return finalValue;
     }
 
-    public List<String> findPathSegments(Dataverse dataverse, List<String> segments) {
-        Dataverse rootDataverse = findRootDataverse();
+    public List<String> findPathSegments(Dataverse dataverse, List<String> segments, Dataverse rootDataverse) {
```

Comment:
Not the part of the task but it would be cleaner if we would use `while` loop instead of recursion. That way we could get rid of `List<String> segments` input parameter